### PR TITLE
[test] simplify test code

### LINKED ISSUE #3065: [test] simplify test code
**What would you like to be added**:
in this commit: https://github.com/iotexproject/iotex-core/commit/892dee9bcf9c7e93efdca1b6c4ade0df21be0e4d#diff-701eab62dfc97c42c827070546fe92f9a280aa70934241aec736cf6402c6be69, `addOneTx()` is a new func to simplify the test code

use it to simplify the following as well:
1. `func addTestingConstantinopleBlocks`
2. `func TestNativeStaking`

**Why is this needed**:
squash lines of test code

### PR DESCRIPTION
**Pull request by LuckyPigeon**
fix #3065

### CODE REVIEW

File: e2etest/native_staking_test.go [hunk 1]
```diff
@@ -99,18 +99,18 @@ func TestNativeStaking(t *testing.T) {
...
-		register2, err := action.SignedCandidateRegister(1, candidate2Name, cand2Addr.String(), cand2Addr.String(),
-			cand2Addr.String(), selfStake.String(), 1, false, nil, gasLimit, gasPrice, cand2PriKey)
-		require.NoError(err)
-
-		fixedTime := time.Unix(cfg.Genesis.Timestamp, 0)
-		require.NoError(ap.Add(context.Background(), register1))
-		require.NoError(createAndCommitBlock(bc, ap, fixedTime))
-		require.NoError(ap.Add(context.Background(), register2))
-		require.NoError(createAndCommitBlock(bc, ap, fixedTime))
+		addOneTx := func(tx action.SealedEnvelope, err error) action.SealedEnvelope {
```
Comment (dustinxie) [inline e2etest/native_staking_test.go:102]: also return the error
`func(tx action.SealedEnvelope, err error) (action.SealedEnvelope, error)`

File: e2etest/native_staking_test.go [hunk 2]
```diff
@@ -99,18 +99,18 @@ func TestNativeStaking(t *testing.T) {
...
+			fixedTime := time.Unix(cfg.Genesis.Timestamp, 0)
```
Comment (dustinxie) [inline e2etest/native_staking_test.go:103]: move this line before `addOneTx`

File: e2etest/native_staking_test.go [hunk 3]
```diff
@@ -99,18 +99,18 @@ func TestNativeStaking(t *testing.T) {
...
+			require.NoError(err)
```
Comment (dustinxie) [inline e2etest/native_staking_test.go:104]: should not use `require` in here
same as https://github.com/iotexproject/iotex-core/pull/3087/files#diff-376031ecf84904d1ad817660ba4a12fcf9143faed75224a178fdad3ebcefd63eR248-R256, and in addition add `createAndCommitBlock`

File: e2etest/native_staking_test.go [hunk 4]
```diff
@@ -303,12 +280,17 @@ func TestNativeStaking(t *testing.T) {
 		require.Equal(voter1BucketIndex, byteutil.BytesToUint64BigEndian(logs[0].Topics[1][24:]))
 		require.Equal(hash.BytesToHash256(cand1Addr.Bytes()), logs[0].Topics[2])
 
+		addOneTx = func(tx action.SealedEnvelope, err error) action.SealedEnvelope {
```
Comment (dustinxie) [inline e2etest/native_staking_test.go:283]: same, see above

File: blockchain/integrity/integrity_test.go [hunk 5]
```diff
@@ -244,13 +244,19 @@ func addTestingConstantinopleBlocks(bc blockchain.Blockchain, dao blockdao.Block
 }
 
 func addTestingTsfBlocks(cfg config.Config, bc blockchain.Blockchain, dao blockdao.BlockDAO, ap actpool.ActPool) error {
+	ctx := context.Background()
+	addOneTx := func(tx action.SealedEnvelope, err error) error {
```
Comment (Liuhaai) [inline blockchain/integrity/integrity_test.go:248]: It is strange to pass `error` as input.
I would prefer to create `addOneTsf` and `addOneExec` separately. The pseudocode code looks like 
```
addOneTsf := func(...) error {
	tsf, err := action.SignedTransfer(...)
	if err != nil {
		return err
	}
	return ap.Add(context.Background(), tsf);
}

```

File: e2etest/native_staking_test.go [hunk 6]
```diff
@@ -362,10 +362,8 @@ func TestNativeStaking(t *testing.T) {
 		require.Equal(hash.BytesToHash256(cand1Addr.Bytes()), logs[0].Topics[2])
 
 		// withdraw	with correct timestamp
-		ws, err = action.SignedReclaimStake(true, 4, selfstakeIndex1, nil, gasLimit, gasPrice, cand1PriKey)
+		ws, err = addOneTx(action.SignedReclaimStake(true, 4, selfstakeIndex1, nil, gasLimit, gasPrice, cand1PriKey))
```
Comment (Liuhaai) [inline e2etest/native_staking_test.go:365]: L368 couldn't be used with current addOneTx(). We can have a workaround or revert it.

**Review by Liuhaai — APPROVED** [review]

**Comment by Liuhaai** [comment]
thx for the contribution

File: e2etest/native_staking_test.go [hunk 7]
```diff
@@ -99,17 +99,23 @@ func TestNativeStaking(t *testing.T) {
...
+				return tx, err
+			}
+			if err := createAndCommitBlock(bc, ap, fixedTime); err != nil {
+				return tx, err
+			}
+			return tx, nil
 		}
 
-		register1 := addOneTx(action.SignedCandidateRegister(1, candidate1Name, cand1Addr.String(), cand1Addr.String(),
+		register1, err := addOneTx(action.SignedCandidateRegister(1, candidate1Name, cand1Addr.String(), cand1Addr.String(),
```
Comment (dustinxie) [inline e2etest/native_staking_test.go:116]: require.NoError(err)

File: e2etest/native_staking_test.go [hunk 8]
```diff
@@ -99,17 +99,23 @@ func TestNativeStaking(t *testing.T) {
...
 			cand1Addr.String(), selfStake.String(), 91, true, nil, gasLimit, gasPrice, cand1PriKey))
-		register2 := addOneTx(action.SignedCandidateRegister(1, candidate2Name, cand2Addr.String(), cand2Addr.String(),
+		register2, err := addOneTx(action.SignedCandidateRegister(1, candidate2Name, cand2Addr.String(), cand2Addr.String(),
```
Comment (dustinxie) [inline e2etest/native_staking_test.go:119]: same, and all other places

**Review by dustinxie — APPROVED** [review]

**Pull request merged**